lib/ohai/plugins/aix/network.rb

```diff
@@ -84,56 +83,52 @@
               netmask = IPAddr.new("255.255.255.255").mask(tmp_prefix.to_i).to_s
             end
 
-            iface[interface][:addresses] ||= Mash.new
-            iface[interface][:addresses][tmp_addr] = { "family" => "inet", "prefixlen" => tmp_prefix }
-            iface[interface][:addresses][tmp_addr][:netmask] = netmask
+            ifaces[int_name][:addresses][tmp_addr] = { "family" => "inet", "prefixlen" => tmp_prefix }
+            ifaces[int_name][:addresses][tmp_addr][:netmask] = netmask
 
-            if lin =~ /broadcast\s(\S+)\s/
-              iface[interface][:addresses][tmp_addr][:broadcast] = $1
+            if line =~ /broadcast\s(\S+)\s/
+              ifaces[int_name][:addresses][tmp_addr][:broadcast] = $1
             end
-          elsif lin =~ %r{inet6 ([a-f0-9\:]+)%?(\d*)/?(\d*)?}
+          elsif line =~ %r{inet6 ([a-f0-9\:]+)%?(\d*)/?(\d*)?}
             # TODO do we have more properties on inet6 in aix? broadcast
-            iface[interface][:addresses] ||= Mash.new
-            iface[interface][:addresses][$1] = { "family" => "inet6", "zone_index" => $2, "prefixlen" => $3 }
+            ifaces[int_name][:addresses] ||= Mash.new
+            ifaces[int_name][:addresses][$1] = { "family" => "inet6", "zone_index" => $2, "prefixlen" => $3 }
           else
-            # load all key-values, example "tcp_sendspace 131072 tcp_recvspace 131072 rfc1323 1"
-            properties = lin.split
-            n = properties.length / 2 - 1
-            (0..n).each do |i|
-              iface[interface][properties[i * 2]] = properties[(i * 2 + 1)]
+            # add all key value data into the interface mash
+            # for example "tcp_sendspace 131072 tcp_recvspace 131072 rfc1323 1"
+            # has keys tcp_sendspace, tcp_recvspace, and rfc1323
+            line.split.each_slice(2) do |key, value|
+              ifaces[int_name][key] = value
             end
           end
         end
       end
 
       # Query macaddress
-      e_so = shell_out("entstat -d #{interface} | grep \"Hardware Address\"")
-      iface[interface][:addresses] ||= Mash.new
+      e_so = shell_out("entstat -d #{int_name} | grep \"Hardware Address\"")
```

Comment:
is the grep still necessary?

Reply:
Nope I have another PR that removes that, but it requires a good amount of spec updates

Reply:
was it ever necessary?

Reply:
:+1:

Reply:
Like many things in these plugins we double check the data. We grep to get 1 line and then for some reason we use a regex to make sure that's in fact the line we got.